core/src/main/java/io/questdb/griffin/engine/groupby/GroupByLong256HashSet.java

```diff
@@ -195,7 +196,9 @@ private long probe(long k0, long k1, long k2, long k3, long index) {
             if (k0Key == k0 && k1Key == k1 && k2Key == k2 && k3Key == k3) {
                 return -index - 1;
             }
-        } while (true);
+        } while (index != index0);
+
+        throw CairoException.critical(0).put("corrupt hash table");
```

Comment:
```suggestion
        throw CairoException.critical(0).put("corrupt long256 hash set state");
```

Reply:
Changed in 20ba60c6b4d95ff2aec27a0f99f028032b891afe